corehq/apps/sms/tests/__init__.py

```diff
@@ -392,4 +414,17 @@ def test_backend(self):
         self.backend3.delete_invoke_doc()
         self.assertFalse(self.backend3.invoke_doc_exists())
 
+    def test_sms_registration(self):
+        incoming("+9991234567", "JOIN {} WORKER tester".format(self.domain), "TEST_CASE_BACKEND")
+        # Test without mobile worker registration enabled
+        self.assertIsNone(CommCareUser.get_by_username("tester@test-domain.commcarehq.org"))
+
+        # Enable mobile worker registration
+        setattr(self.domain_obj, "sms_mobile_worker_registration_enabled", True)
+        self.domain_obj.save()
+
+        incoming("+9991234567", "JOIN {} WORKER tester".format(self.domain), "TEST_CASE_BACKEND")
+        self.assertIsNotNone(CommCareUser.get_by_username("tester@test-domain.commcarehq.org"))
 
+        # Test a duplicate registration
+        self.assertTrue(incoming("+9991234568", "JOIN {} WORKER tester".format(self.domain), "TEST_CASE_BACKEND"))
```

Comment:
Initially I included this line as I was previously raising an error and this proved I was not.

The intended outcome of registering twice is sending an return SMS.  In other places I see you send the SMS using the TestCaseBackend, but I am not sending the SMS from the test.  Is there an appropriate way to ensure that SMS is triggered?